README.md

```diff
@@ -30,9 +31,24 @@ Official pure Rust typed client for ClickHouse DB.
 
 Note: [ch2rs](https://github.com/ClickHouse/ch2rs) is useful to generate a row type from ClickHouse.
 
+## Validation
+
+Starting from 0.14.0, the crate supports validation of the row types against the ClickHouse schema, as
+`RowBinaryWithNamesAndTypes` format is used by default instead of `RowBinary`. Additionally, with enabled validation, 
+the crate supports structs with correct field names and matching types, but incorrect order of the fields, 
+with a slight (5-10%) performance penalty.
+
+If you want to disable validation entirely, essentially reverting the client behavior to pre-0.14.0, you can use
+`Client::with_validation(false)`, which will switch the fetch format to `RowBinary` instead.
+
+Depending on the dataset, disabling validation can yield from x1.1 to x3 performance improvement, 
+but it is not recommended to use it in production, as it can lead to unclear runtime errors 
```

Comment:
There are no problems using it if you have (at least) a simple smoke test. I suggest avoiding any "recommendation" because it's already a default behavior, so telling people how to increase performance is enough if they have a simple test.